storage/raftstorage.go

```diff
@@ -67,80 +67,88 @@ func (r *Range) InitialState() (raftpb.HardState, raftpb.ConfState, error) {
 	return hs, cs, nil
 }
 
-// loadLastIndex looks in the engine to find the last log index.
+// setLastIndex persists a new last index and updates its cached
+// value in the range. If a non-nil engine is passed, it is used for the
+// write; otherwise the range's default engine is used. Commit() will
```

Comment:
Not fond of this auto-committing behaviour but not having that lead to code duplication and I'd really prefer having one callsite for this only. Thoughts?
Same for setAppliedIndex.